### ISSUE TITLE
focus problem with VoiceOver

### ISSUE THREAD

**Issue by Veyfeyken** [scottaohara/accessible_modal_window#1]
Hi, great work.

Some feedback regarding screenreaders:
- Works fine with NVDA in Firefox
- Jaws (in IE11) doesn't read the aria-label for the close button of the first dialog:

`<button class="modal__outro__close" type="button" data-modal-close="" aria-label="Close Modal" aria-controls="modal_1">`
- VoiceOver (Mac OS X 10.11.6 with Safari 9.1.3) first gives focus to the modal and then the close button while navigating with the VoiceOver keys. The content itself is skipped. You have to navigate backwards to read the actual content. 
- The focus doesn't return to the button that opened the dialog after you close it with VoiceOver. The focus stays on the close button (which is no longer visible) and gets stuck. It's no longer possible to navigate with the VoiceOver keys. This happened several times but strangely not every time.

**Comment by scottaohara** [comment]
Hey @Veyfeyken thanks for opening the issue with this.

Good to know about the Jaws issue. I'll get on fixing that right away.

Per the VoiceOver stuff.  The first issue you mentioned is actually an issue I've seen a lot with forced focus and VoiceOver navigation in general.  I have a separate article I've been writing about that itself that I hope to release soon, as from what I understand in my testing with VoiceOver users it's a common issue.

I didn't experience the last issue with the focus not returning to the original trigger in VoiceOver.  I however was testing on OSX 10.10.5 / Safari 9.1.3, and I know Apple made some updates to VoiceOver between 10.10 and 10.11.  I'll look into this again too.


Thanks again!

**Comment by bart-simons** [comment]
The reported JAWS problem occurs in version 15 and 16. In version 17 the button label is read correctly

**Comment by bart-simons** [comment]
Could you explain why you added aria-controls to the close button of the modal windows? I would suggest to leave the attribute out. Also see http://www.heydonworks.com/article/aria-controls-is-poop

**Comment by scottaohara** [comment]
Hi @bart-simons 

Thanks for sending over the link to Heydon's article.  I had previously read it, and am aware of the poop that is aria-controls.  I could definitely see removing it due to it's poor support, but if it actually worked the way it was intended, it would be appropriate on the close button, as it is having the inverse effect of the open trigger (which also uses aria-controls).

[Léonie Watson's comments to Heydon's article](http://www.heydonworks.com/article/aria-controls-is-poop#comment-2882197520) do give good reasons as to why aria-controls is important to fix and how it will be beneficial once it's consistently implemented.

Anyway, thank you for your suggestions and your additional testing w/JAWS per the aria-label issue.

**Comment by scottaohara** [comment]
while updating the modal some more, i retested against some of the focus issues in voice over w/ osx 10.10.5 & Safari 10.1.1 and latest VO + mobile Safari, and wasn't able to reproduce any of the issues.

with that in mind, i'm going to close the issue for now, but will continue to do testing on other VO/Safari combos.

thanks again for reporting this.
